fastlane/lanes/screenshots.rb

```diff
@@ -302,30 +261,44 @@
 
 
   #####################################################################################
-  # rebuild_screenshot_devices
+  # upload_and_replace_screenshots_in_play_store
   # -----------------------------------------------------------------------------------
-  # This lane rebuilds all of the emulators used for generating screenshots. Beware – running
-  # this action will overwrite the following emulators:
-  #     - Nexus 9 API 28
-  #     - Pixel 2 XL API 28
-  # It will not overwrite any other devices.
+  # This lane uploads the screenshots in /metadata/android/{locale}/images to Play
+  # Store and replaces the existing ones.
+  # If a locale doesn't have any screenshots, it'll be skipped.
   # -----------------------------------------------------------------------------------
   # Usage:
-  # fastlane rebuild_screenshot_devices
+  # bundle exec fastlane upload_and_replace_screenshots_in_play_store app:<wordpress|jetpack>
   #
   # Example:
-  # fastlane rebuild_screenshot_devices
+  # bundle exec fastlane upload_and_replace_screenshots_in_play_store app:wordpress
   #####################################################################################
-  desc "Rebuild screenshot devices"
-  lane :rebuild_screenshot_devices do |options|
-    emulators = [
-        Dir.pwd + "/emulators/Nexus_9_API_28.ini",
-        Dir.pwd + "/emulators/Pixel_2_XL_API_28.ini",
-        Dir.pwd + "/emulators/Pixel_3_API_28.ini",
-    ]
-
-    emulators.each do |emulator_configuration|
-        sh("helpers/copy-device.sh '#{emulator_configuration}'")
+  desc 'Upload Screenshots to Play Store and Replaces the existing ones'
+  lane :upload_and_replace_screenshots_in_play_store do |options|
+    app = get_app_name_option!(options)
+    package_name = APP_SPECIFIC_VALUES[app.to_sym][:package_name]
+    metadata_dir = File.join('fastlane', APP_SPECIFIC_VALUES[app.to_sym][:metadata_dir], 'android')
+
+    upload_to_play_store(
+      package_name: package_name,
+      metadata_path: metadata_dir,
+      skip_upload_apk: true,
+      skip_upload_aab: true,
+      skip_upload_metadata: true,
+      skip_upload_changelogs: true,
+      skip_upload_images: true,
+      skip_upload_screenshots: false,
+      json_key: UPLOAD_TO_PLAY_STORE_JSON_KEY
+    )
+  end
+
+
+  # Rebuilds all of the emulators used for generating screenshots.
+  #
+  desc "Rebuild emulators used for screenshots"
+  lane :create_emulators do |options|
+    SCREENSHOT_DEVICES.each do |device|
+      create_avd(api: device[:api], device: device[:device])
```

Comment:
TODO: We need to call `install_system_image(api: device[:api])` first here.

Though in practice we might as well get rid of that lane, now that I've refactored the `screenshots` lane to directly call `create_avd` on its own loop — in order to only create the emulators for which screenshots are requested to be build, instead of all of them even the ones we filtered out — instead of having the `screenshots` lane call `create_emulators` at its start like before.

Reply:
Fixed in d418d86d3c67b1e0355c98b5f3735b4dea07ae4f